src/hotspot/os/bsd/globals_bsd.hpp

```diff
@@ -28,13 +28,16 @@
 //
 // Declare Bsd specific flags. They are not available on other platforms.
 //
-#define RUNTIME_OS_FLAGS(develop,     \
-                         develop_pd,  \
-                         product,     \
-                         product_pd,  \
-                         notproduct,  \
-                         range,       \
-                         constraint)
+#define RUNTIME_OS_FLAGS(develop,                                       \
+                         develop_pd,                                    \
+                         product,                                       \
+                         product_pd,                                    \
+                         notproduct,                                    \
+                         range,                                         \
+                         constraint)                                    \
+                                                                        \
+  AARCH64_ONLY(develop(bool, WXCheckAtSafepoint, false,                 \
```

Comment:
Why not enable always as this is a develop build? Do we even need to manage this on a flag?

Also not really "safepoint" as the check also applies to handshakes.

Reply:
Even in debug builds the check may be to assertive. This checks exposes 22 failures in tier1 in jfr, gc, compiler, serviceability, and foreign (hope to be fixed soon), and they look like actual possible issues. So I suggest to have the option but keep it disabled.